getnacelle/nacelle-js
# feat: init sanity-plugin-nacelle

clones and modifies sanity-plugin-pim-linker for v2 data



## WHY are these changes introduced?

Addresses [ENG-4200](https://nacelle.atlassian.net/browse/ENG-4200) - sets up the v2 version of the Sanity plugin with it's new name. This will allow all v2 Sanity merchants to link their v2 products to their Sanity content. 



## WHAT is this pull request doing?

Note, this package is a fork of [`@nacelle/sanity-plugin-pim-linker`](https://github.com/getnacelle/nacelle-sanity-plugin-pim-linker). See [PR #10 in that repo](https://github.com/getnacelle/nacelle-sanity-plugin-pim-linker/pull/10) for a diff of _most_ (but not all) of the changes.

1. Creates the package `@nacelle/sanity-plugin-nacelle-input` from the old `@nacelle/sanity-plugin-pim-linker` and makes updates where needed to support the new name. 
2. Uses the search API to query v2 spaces (thanks to @KrisQ) and populate that data. Also removes all in-memory filtering/paginating to make it more performant for large spaces.
3. Updates the config to take a `nacelleEndpoint` and `nacelleSpaceToken` (or an array of objects featuring both fields and a `spaceName` to better match V2 requirements. 




## How to Test

🖥 [Local development instructions](#link-to-repo-readme)
🗒 [General testing guidelines](#link-to-testing-guidelines)
📄 [Changelog guidelines](#link-to-changelog-guidelines)

1. Clone and setup this branch of `nacelle-js` (`gh pr checkout #165` if using the GH cli)
2. Run `npm i && npm run bootstrap` from the root of this repo.
3. `cd packages/sanity-plugin-nacelle-input`
4. Run `npm run build`
5. `npm link`
6. Clone or create a Sanity workspace and open it in a terminal.
7. If it already has `@nacelle/sanity-plugin-pim-linker` installed, uninstall it with `npm rm @nacelle/sanity-plugin-pim-linker` & remove `@nacelle/sanity-plugin-pim-linker` from the `sanity.json` plugins array. 
8. Install this plugin by running `npm link @nacelle/sanity-plugin-nacelle-input`
9. Add `@nacelle/sanity-plugin-nacelle-input` to the plugins array in the `sanity.json` file. 
10. Follow the instructions for setting up this package & it's credentials outlined in the [README](/packages/sanity-plugin-nacelle-input/README.md#credentials)
11. Test your input by creating/editing a content entry with the `nacelleData` field and searching for products and productCollections.
   - If you use the array of spaces, you should also have a select input above your search input to allow you to choose which space to search for products/product collections.

### Testing checklist

* [ ] Tested for accessibility
* [ ] Updated public facing documentation
